src/cryptonote_core/loki_name_system.cpp

```diff
@@ -463,63 +463,87 @@ static bool check_condition(bool condition, std::string* reason, T&&... args) {
   return condition;
 }
 
-bool validate_lns_name(mapping_type type, std::string const &name, std::string *reason)
+bool validate_lns_name(mapping_type type, std::string name, std::string *reason)
 {
   std::stringstream err_stream;
   LOKI_DEFER { if (reason) *reason = err_stream.str(); };
 
-  size_t max_name_len = 0;
-  if (type == mapping_type::session)      max_name_len = lns::SESSION_DISPLAY_NAME_MAX;
+  bool const is_lokinet = is_lokinet_type(type);
+  size_t max_name_len   = 0;
+
+  if (is_lokinet)                         max_name_len = lns::LOKINET_DOMAIN_NAME_MAX;
+  else if (type == mapping_type::session) max_name_len = lns::SESSION_DISPLAY_NAME_MAX;
   else if (type == mapping_type::wallet)  max_name_len = lns::WALLET_NAME_MAX;
-  else if (is_lokinet_type(type))         max_name_len = lns::LOKINET_DOMAIN_NAME_MAX;
   else
   {
     if (reason) err_stream << "LNS type=" << type << ", specifies unhandled mapping type in name validation";
     return false;
   }
 
   // NOTE: Validate name length
-  if (check_condition((name.empty() || name.size() > max_name_len), reason, "LNS type=", type, ", specifies mapping from name -> value where the name's length=", name.size(), " is 0 or exceeds the maximum length=", max_name_len, ", given name=", name))
+  name = tools::lowercase_ascii_string(name);
```

Comment:
If you use `name = lowercase(std::move(name));` here you can save a string reallocation.

Reply:
Added